components/public-api/gitpod/v1/envvar.proto

```diff
@@ -0,0 +1,88 @@
+syntax = "proto3";
+
+package gitpod.v1;
+
+option go_package = "github.com/gitpod-io/gitpod/components/public-api/go/v1";
+
+service EnvironmentVariableService {
+  // ListEnvironmentVariables returns all environment variables for the
+  // authenticated user.
+  //
+  // +return NOT_FOUND User does not have access to a workspace with the given
+  // ID +return NOT_FOUND Workspace does not exist
+  rpc ListEnvironmentVariables(ListEnvironmentVariablesRequest)
+      returns (ListEnvironmentVariablesResponse) {}
+
+  // UpdateEnvironmentVariable updates an environment variable for the
+  // authenticated user.
+  rpc UpdateEnvironmentVariable(UpdateEnvironmentVariableRequest)
+      returns (UpdateEnvironmentVariableResponse) {}
+
+  // CreateEnvironmentVariable creates a new environment variable for the
+  // authenticated user.
+  rpc CreateEnvironmentVariable(CreateEnvironmentVariableRequest)
+      returns (CreateEnvironmentVariableResponse) {}
+
+  // DeleteEnvironmentVariable deletes an environment variable for the
+  // authenticated user.
+  rpc DeleteEnvironmentVariable(DeleteEnvironmentVariableRequest)
+      returns (DeleteEnvironmentVariableResponse) {}
+}
+
+message UserScope {
+  string repository_pattern = 1;
+}
+
+message ConfigurationScope {
+  string configuration_id = 1;
+  bool censored = 2;
+}
+
+message EnvironmentVariable {
+  string id = 1;
+  string name = 2;
+  string value = 3;
+  oneof scope {
+    UserScope user_scope = 4;
+    ConfigurationScope configuration_scope = 5;
+  }
+}
+
+message ListEnvironmentVariablesRequest {
+  string workspace_id = 1;
+  string configuration_id = 2;
```

Comment:
can we add docs to whether these arguments are mandatory and how they are used?

Reply:
Not sure about this yet, see https://github.com/gitpod-io/gitpod/pull/19067#discussion_r1393701886